great_expectations/rule_based_profiler/data_assistant/data_assistant.py

```diff
@@ -150,38 +152,38 @@ def _build_profiler(self) -> None:
             )(**expectation_kwargs)
             variables.update(convert_variables_to_dict(variables=profiler.variables))
             rules = profiler.rules
-            for rule in rules:
-                domain_builder = rule.domain_builder
-                parameter_builders = rule.parameter_builders or []
-                parameter_builders.extend(
-                    self.metrics_parameter_builders_by_domain[
-                        Domain(
-                            domain_builder.domain_type,
-                        )
-                    ]
-                )
-                expectation_configuration_builders = (
-                    rule.expectation_configuration_builders or []
-                )
-                self.profiler.add_rule(
-                    rule=Rule(
-                        name=rule.name,
-                        variables=rule.variables,
-                        domain_builder=domain_builder,
-                        parameter_builders=parameter_builders,
-                        expectation_configuration_builders=expectation_configuration_builders,
-                    )
-                )
+            self._add_rules_to_profiler(rules=rules)
+
+        common_rule_names: Set[str] = {
+            rule.name for rule in (self.profiler.rules or {})
+        } & {rule.name for rule in (self.rules or {})}
+        if common_rule_names:
+            raise ge_exceptions.ProfilerConfigurationError(
+                message=f"""Rule names in {self.__class__.__name__} must be unique; duplicate(s) found \
+({common_rule_names}).
+"""
+            )
 
+        self._add_rules_to_profiler(rules=self.rules)
+
+        variables.update(self.variables or {})
```

Comment:
Same note here. Could we just do a check to see if `self.variables` is present and only update if so? Very minor change but I think it improves readability.

I feel less strongly about this than the prior comment so I'll leave it up to you.

Reply:
@cdkini Fixing (in next commit).  Thanks!